tests/workspace_smoke/doc_demo_smoke.py

```
@@ -0,0 +1,571 @@
+        "source /home/user/.bashrc; "
+        f"{package_setup}"
+        f"timeout --preserve-status {demo.timeout_seconds:d}s {demo.command}"
+    )
+    command = [*docker_base(demo, args), "-lc", shell_command]
+
+    log_path.parent.mkdir(parents=True, exist_ok=True)
+    print(f"$ {' '.join(command)}")
+    print(f"log: {display_path(log_path)}")
+    with log_path.open("w", encoding="utf-8") as log_file:
+        log_file.write(f"$ {' '.join(command)}\n")
+        process = subprocess.Popen(command, cwd=REPO_ROOT, text=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)